### PR TITLE
Make sure pgbouncer-exporter docker image is linux/amd64

### LINKED ISSUE #11145: airflow-pgbouncer-exporter image : standard_init_linux.go:211: exec user process caused "exec format error"
**Apache Airflow version**: I'm using the helm chart that is in master branch

**What happened**:
I'm having trouble with the airflow helm chart, when pgbouncer is deployed `pgbouncer.enabled=true` the pgbouncer deployment fails because one of the container is the pod is continuously crashing. 

I've narrowed down the problem to: 

docker run -it --rm  apache/airflow:airflow-pgbouncer-exporter-2020.09.05-0.5.0
standard_init_linux.go:211: exec user process caused "exec format error"

**What you expected to happen**:

It should run the `pgbouncer_exporter server`.

**How to reproduce it**:

docker run -it --rm  apache/airflow:airflow-pgbouncer-exporter-2020.09.05-0.5.0
standard_init_linux.go:211: exec user process caused "exec format error"

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by ecerulm**
Previously if the `./build_and_push.sh` was executed from a macOS machine it will upload a docker image containing a macOS executable. This image will fail to run  with 

```
docker run -it --rm apache/airflow:airflow-pgbouncer-exporter-2020.09.05-0.5.0
standard_init_linux.go:211: exec user process caused "exec format error"
```

Now the `./build_and_push.sh` does NOT use the local go compiler instead it compiles using the golang Docker image so that the generated executable is linux/amd64. This also removes the need to have a specific go version installed in the build machine. 

Closes #11145

### CODE REVIEW

**Review by potiuk — APPROVED** [review]
Nice!. Indeed I tested it on linux only :). Good catch!

File: chart/dockerfiles/pgbouncer-exporter/Dockerfile
Comment (potiuk) [inline chart/dockerfiles/pgbouncer-exporter/Dockerfile]: This one should be removed though. Maitainer directive is deprecated and we replaced it with the maintainer label which is the recommended approach: https://docs.docker.com/engine/reference/builder/#maintainer-deprecated

**Review by potiuk — CHANGES_REQUESTED** [review]
Nice!. Indeed I tested it on linux only :). Good catch!

**Review by potiuk — APPROVED** [review]

**Comment by ecerulm** [comment]
@potiuk Please note that the actual image currently in Docker Hub for linux/amd64 actually contains a macOS executable as I explained in #11145. So it's important that you or anybody that can push the docker image rerun the `./build_and_push.sh` to replace the **currently unusable docker image**.

**Comment by ecerulm** [comment]
@potiuk I did more more modifications and squash all commits. 

I have some insights on why the [Docker Hub for apache/airflow:airflow-pgbouncer-exporter-2020.09.05-0.5.0](https://hub.docker.com/layers/apache/airflow/airflow-pgbouncer-exporter-2020.09.05-0.5.0/images/sha256-ae731aeef11cd759633e899ceddc4d5e79cead91ef221deeaeaf66d7b757e2eb?context=explore) shows ` MAINTAINER Juraj Bubniak <juraj.bubniak@gmail.com>`. 

It seems that the original `build_and_push` did `cd pgbouncer_exporter-0.5.0`, compiled pgbouncer and then executed `docker build . ` **in the `pgbouncer_exporter-0.5.0`  thus using the pgbouncer_exporter's  Dockerfile instead of "our" [Dockerfile](https://github.com/apache/airflow/blob/master/chart/dockerfiles/pgbouncer-exporter/Dockerfile). 

I've fixed that in the latest version.

**Comment by potiuk** [comment]
@ecerulm - Would you mind also changing the date at the top to today ( and refer to it from the helm chart ?) then i will push the image just before merging it.

**Comment by potiuk** [comment]
The version should be updated in separate PR, the image should be pushed before to apache registry

**Pull request merged**
